### DISCUSSION TITLE
Can GameHub only run a single added app or game at a time?

### DISCUSSION THREAD

**Discussion by AlfredoRodda** [tkashkin/GameHub#582] [Q&A] [answered]
Hi,

Just to clarify, I do not mean trying to run two instances of GameHub or two instance of the same app or game through GameHub.  It seems once you have an app or game running within GameHub, you can not run another different one until the first is closed. 

Is there a way to run more than one added app or game at a time? 

Steps to reproduce.

1. Launch GameHub
2. Run an app or game. (The banner will say "Running")
3. Try to run another app or game by double clicking its respective banner or by right clicking and pressing "Run" which is not clickable.

GameHub Version - 0.16.1-8-dev

**Comment by tkashkin** [comment, marked as answer]
Currently GameHub can only run one game at a time and this limit is not configurable.

You can change this by removing the check at https://github.com/tkashkin/GameHub/blob/master/src/data/Runnable.vala#L81 and rebuilding GameHub from source. You may also want to remove other references to `Runnable.IsLaunched` and `Steam.IsAnyAppRunning`.

**Reply by AlfredoRodda** [reply]
Will this behavior always be the default? Is there a possibility to allow running multiple games in a future release?

Thank you for your response and suggestion.

**Reply by tkashkin** [reply]
It could be configurable (probably not in UI, only via `dconf`), but I'm not really sure it needs to be.

Can you explain your usecase for launching multiple games at the same time? Can it be useful for other users and what should be a default in your opinion?

**Reply by AlfredoRodda** [reply]
I just tend to find myself have multiple games or emulators open at once. Sort of like how some have multiple tabs or apps running. In general, I just though it was typical to have the ability to run multiple apps at once. Below are a couple of examples but this is not an issue at all, it is very minor.


- If I launch a game and it has an update and I would rather play another instead of waiting before I can launch something.
- If a game is running and I want to "pause" that and open up another.

**Reply by tkashkin** [reply]
Yes, I guess the limit is not needed here. I've created #583 for this.

The limit was added to fix the same game launching multiple times (#280, #281), but GameHub should probably allow to launch multiple *different* games at the same time.

**Reply by AlfredoRodda** [reply]
No worries, if this behavior is eventually changed in the future where multiple games are allowed to run then great! If not it is a minor inconvenience, and not an issue. Thank you for your time.
